src/executor/attach_op_execs_pass.cc

```diff
@@ -23,42 +23,85 @@ const OperatorProperty* OpPropGetOpProperty(const NodeAttrs& attrs);
 
 namespace exec {
 
-// stateful compute executor
-class StatefulComputeExecutor : public OpExecutor {
+// abstract OpExecutor which provides storage fallback procedure on
+// non-default inputs and outputs
+class StorageFallbackOpExecutor : public OpExecutor {
  public:
-  void Run(RunContext rctx, bool is_gpu) override {
+  explicit StorageFallbackOpExecutor(const std::vector<uint32_t> &mutate_idx)
+      : mutate_idx_(mutate_idx) {}
+
+  void Setup() override {
+    using namespace common;
+    in_data_.clear(); out_data_.clear();
+    pre_temp_src_.clear(); pre_temp_dst_.clear();
+    post_temp_src_.clear(); post_temp_dst_.clear();
+    in_temp_idx_map_.clear();
+    SetupDefaultBlobs(in_array, &in_data_, &pre_temp_src_, &pre_temp_dst_, &in_temp_idx_map_);
+    SetupDefaultBlobs(out_array, &out_data_, &post_temp_dst_, &post_temp_src_);
+    for (const auto idx : mutate_idx_) {
+      if (in_temp_idx_map_.find(idx) != in_temp_idx_map_.end()) {
+        post_temp_src_.push_back(pre_temp_dst_[in_temp_idx_map_[idx]]);
```

Comment:
`in_temp_idx_map_[idx]` was found in the line above. Use an iterator to store the found result and deference the iterator here to replace `in_temp_idx_map_[idx]` for avoiding searching the map for the second time.